pyodide-build/pyodide_build/tests/test_pypi.py

```diff
@@ -0,0 +1,321 @@
+            )
+    output_dir = tmp_path / "dist"
+    # one pure-python package (doesn't need building) which depends on one sdist package (needs building)
+    pkgs = ["eth-hash[pycryptodome]==0.5.1", "safe-pysha3 (>=1.0.0)"]
+
+    os.chdir(tmp_path)
+    for p in pkgs:
+        result = runner.invoke(
+            build.app,
+            [p, "--build-dependencies"],
+        )
+        assert result.exit_code == 0, result.stdout
+
+    built_wheels = set(output_dir.glob("*.whl"))
+    assert len(built_wheels) == 3
+
+
+def test_fake_pypi_succeed(tmp_path, fake_pypi_url):
+    if "dev" in pyodide_build_version:
+        if "EMSDK" not in os.environ or "PYODIDE_ROOT" not in os.environ:
+            pytest.skip(
+                reason="Can't build recipe in dev mode without building pyodide first. Skipping test"
+            )
+    output_dir = tmp_path / "dist"
+    # build package that resolves right
+    with work_in_dir(tmp_path):
+        result = runner.invoke(
+            build.app,
+            ["resolves-package", "--build-dependencies"],
+        )
+
+        assert result.exit_code == 0, str(result.stdout) + str(result)
+
+    built_wheels = set(output_dir.glob("*.whl"))
+    assert len(built_wheels) == 5
+
+
+def test_fake_pypi_resolve_fail(tmp_path, fake_pypi_url):
+    if "dev" in pyodide_build_version:
+        if "EMSDK" not in os.environ or "PYODIDE_ROOT" not in os.environ:
+            pytest.skip(
+                reason="Can't build recipe in dev mode without building pyodide first. Skipping test"
+            )
+    output_dir = tmp_path / "dist"
+    # build package that resolves right
+
+    with work_in_dir(tmp_path):
+        result = runner.invoke(
+            build.app,
+            ["fails-package", "--build-dependencies"],
+        )
+
+    # this should fail
+    assert result.exit_code != 0, result.stdout
+    built_wheels = set(output_dir.glob("*.whl"))
+    assert len(built_wheels) == 0
+
+
+def test_fake_pypi_extras_build(tmp_path, fake_pypi_url):
+    if "dev" in pyodide_build_version:
+        if "EMSDK" not in os.environ or "PYODIDE_ROOT" not in os.environ:
+            pytest.skip(
+                reason="Can't build recipe in dev mode without building pyodide first. Skipping test"
+            )
+    output_dir = tmp_path / "dist"
+    # build package that resolves right
+
+    with work_in_dir(tmp_path):
+        result = runner.invoke(
+            build.app,
+            ["pkg-b[docs]", "--build-dependencies"],
+        )
+
+    # this should fail
+    assert result.exit_code == 0, result.stdout
+    built_wheels = set(output_dir.glob("*.whl"))
+    assert len(built_wheels) == 2
+
+
+@pytest.mark.parametrize(
+    "fn", [test_fetch_or_build_pypi, test_fetch_or_build_pypi_with_deps_and_extras]
+)
+def test_host_in_node_test(runtime, fn, tmp_path):
+    if runtime.startswith("node"):
+        fn(tmp_path)
```

Comment:
These tests are not actually running. I am trying to fix broken CLI in #3390. Probably you will need to fix tests in a same way in #3390.